### ISSUE TITLE
Add Jacoco and uploads test results to codecov

### ISSUE THREAD

**Issue by marandaneto** [getsentry/sentry-android-gradle-plugin#87]
See Java repo https://app.codecov.io/gh/getsentry/sentry-java

**Comment by cortinico** [comment]
> when doing this, we can remove the `Upload Test Results` step from `pre-merge.yaml`

The reason behind that step is to be able to easily see what was the failure in the tests. I'm not sure uploading code coverage to codecov will include also test results (like stacktraces and so on).

**Comment by marandaneto** [comment]
yeah they are not really related, my bad, this issue is more about the `codecov`

**Comment by romtsn** [comment]
we're not using codecov on this repo anymore, closing
